programs/manifest/src/program/processor/swap.rs

```diff
@@ -88,26 +91,43 @@ pub(crate) fn process_swap(
     // net token transfers will be handled later
     dynamic_account.deposit(payer.key, in_atoms, is_base_in)?;
 
+    // 4 cases:
+    // 1. Exact in base. Simplest case, just use the base atoms given.
+    // 2. Exact in quote. Search the asks for the number of base atoms in bids to match.
+    // 3. Exact out quote. Search the bids for the number of base atoms needed to match to get the right quote out.
+    // 4. Exact out base. Use the number of out atoms as the number of atoms to place_order against.
     let base_atoms: BaseAtoms = if is_exact_in {
         if is_base_in {
             // input=desired max(base) output=checked min(quote)
             BaseAtoms::new(in_atoms)
         } else {
             // input=desired max(quote) output=checked min(base)
-            // round down base amount to not cross quote limit
-            dynamic_account.impact_base_atoms(true, false, QuoteAtoms::new(in_atoms))?
+            dynamic_account.impact_base_atoms(true, QuoteAtoms::new(in_atoms))?
```

Comment:
this rounding difference was actually important, i'm not sure it can go

Reply:
it was fully defined by is_bid. wasnt a case of is_bid != round_up